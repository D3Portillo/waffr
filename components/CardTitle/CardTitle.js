import React, { Fragment } from "react";
import {
  useEtherBalance,
  useEthers,
  useToken,
  useTokenBalance,
} from "@usedapp/core";

import { localizeEthBalance } from "@/lib/utils/ether";

const SafeSymbolRender = ({ token }) => {
  const state = useToken(token);
  return state && state.symbol;
};

function CardTitle({ withAccountInfo, children, withToken }) {
  const { account } = useEthers();
  const defaultBalance = useEtherBalance(account);
  const tokenBalance = useTokenBalance(withToken, account);
  const balance = withToken ? tokenBalance : defaultBalance;
  const formatedBalance = localizeEthBalance(balance);
  const symbol =
    withToken && balance ? <SafeSymbolRender token={withToken} /> : "ETH";
  return (
    <div className="flex-col">
      <div className="text-4xl pb-2">
        <strong className="flex items-center">
          {withToken && symbol} {children}
        </strong>
      </div>
      {withAccountInfo && (
        <Fragment>
          <div className="px-1 text-white text-opacity-30">
            <b>YOUR ADDRESS:</b> {account || "NO CONNECTION"}
          </div>
          <div className="px-1 text-white text-opacity-30">
            <b>BALANCE:</b> {formatedBalance} {symbol}
          </div>
        </Fragment>
      )}
    </div>
  );
}

CardTitle.defaultProps = {
  withAccountInfo: false,
  withToken: false,
  children: null,
};

export default CardTitle;
