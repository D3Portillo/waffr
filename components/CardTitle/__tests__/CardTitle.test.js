import { render } from "@testing-library/react";
import * as daap from "@usedapp/core";
/** @type { import("jest.setup").MOCKED_DAAP } */
const usedaap = daap

import CardTitle from "../CardTitle";

describe("CardTitle", () => {
  let node;
  afterEach(() => {
    if (node && node.unmount) {
      node.unmount();
    }
    jest.clearAllMocks();
  });

  it("matches snapshot", () => {
    node = render(<CardTitle>CONTENT</CardTitle>);
    expect(node.container).toMatchSnapshot();
  });

  it("renders balance, account withAccountInfo=true", () => {
    node = render(<CardTitle withAccountInfo>CONTENT</CardTitle>);
    const { container } = node;
    expect(container.textContent).toMatch(/BALANCE:.*1.*ETH/g);
    expect(container.textContent).toMatch(/YOUR ADDRESS:.*NO CONNECTION/g);
  });

  it("renders symbol when withToken=MYTOKEN", () => {
    usedaap.useToken.mockImplementation((symbol) => ({ symbol }));
    const MYTOKEN = "MYTOKEN";
    node = render(<CardTitle withToken={MYTOKEN}>CONTENT</CardTitle>);
    const { container } = node;
    expect(container).toMatchSnapshot();
    expect(container.textContent).toMatch(MYTOKEN);
  });
});
