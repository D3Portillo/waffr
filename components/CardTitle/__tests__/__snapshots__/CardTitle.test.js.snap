// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`CardTitle matches snapshot 1`] = `
<div>
  <div
    class="flex-col"
  >
    <div
      class="text-4xl pb-2"
    >
      <strong
        class="flex items-center"
      >
         
        CONTENT
      </strong>
    </div>
  </div>
</div>
`;

exports[`CardTitle renders symbol when withToken=MYTOKEN 1`] = `
<div>
  <div
    class="flex-col"
  >
    <div
      class="text-4xl pb-2"
    >
      <strong
        class="flex items-center"
      >
        MYTOKEN
         
        CONTENT
      </strong>
    </div>
  </div>
</div>
`;
